test/Schemes/ProjectiveSchemes.jl

```diff
@@ -110,3 +110,143 @@ end
   @test_broken as_covered_scheme(P2)  # too exotic
 end
 
+@testset "morphisms of projective schemes I" begin
+  R, (x,y) = QQ["x", "y"]
+
+  IP2_X = projective_space(R, 2, var_name="u")
+  projective_space(R, ["u", "v", "w"])
+  X = base_scheme(IP2_X)
+  projective_space(X, ["u", "v", "w"])
+
+  inc = ClosedEmbedding(X, ideal(OO(X), [x^2 - y^2]))
+  Y = domain(inc)
+
+  IP2_Y = projective_space(Y, 2, var_name="v")
+  projective_space(Y, ["u", "v", "w"])
+  projective_space(OO(Y), ["u", "v", "w"])
+
+  U = PrincipalOpenSubset(X, x)
+
+  IP2_U = projective_space(OO(U), 2, var_name="u")
+  projective_space(U, ["u", "v", "w"])
+  projective_space(OO(U), ["u", "v", "w"])
+  set_base_scheme!(IP2_U, U)
```

Comment:
So why not directly `IP2_2 = projective_space(U,2 ,var_name"u")`? Then the base_scheme is set already?
Or is this just a test for the setter?

Reply:
It's a test for the setter.